pkg/resources/server.go

```diff
@@ -5,34 +5,97 @@ import (
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	registerapi "k8s.io/kubernetes/pkg/kubelet/apis/pluginregistration/v1"
 	"log"
 	"net"
 	"os"
 	"path/filepath"
 	"strconv"
 	"time"
-
-	"golang.org/x/net/context"
-	"google.golang.org/grpc"
-	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
-	registerapi "k8s.io/kubernetes/pkg/kubelet/apis/pluginregistration/v1"
 )
 
-// resourceServer implements the Kubernetes device plugin API
+type resourcesServerConnector struct {
+	server *grpc.Server
+}
+
 type resourceServer struct {
-	resourceName string
-	socketName   string
-	socketPath   string
-	watchMode    bool
-	devs         []*pluginapi.Device
-	deviceSpec   []*pluginapi.DeviceSpec
-	stop         chan interface{}
-	stopWatcher  chan bool
-	health       chan *pluginapi.Device
-	server       *grpc.Server
+	resourceName  string
+	watchMode     bool
+	endPoint      string
+	socketPath    string
+	devs          []*pluginapi.Device
+	deviceSpec    []*pluginapi.DeviceSpec
+	stop          chan interface{}
+	stopWatcher   chan bool
+	health        chan *pluginapi.Device
+	socketWatcher *fsnotify.Watcher
+	rsConnector   types.ResourceServerConnector
+}
+
+func (rsc *resourcesServerConnector) GetServer() *grpc.Server {
+	return rsc.server
+}
+
+func (rsc *resourcesServerConnector) CreateServer() {
+	rsc.server = grpc.NewServer([]grpc.ServerOption{}...)
+}
+
+func (rsc *resourcesServerConnector) DeleteServer() {
+	rsc.server = nil
+}
+
+func (rsc *resourcesServerConnector) Listen(socketType, socketPath string) (net.Listener, error) {
+	return net.Listen(socketType, socketPath)
+}
+
+func (rsc *resourcesServerConnector) Serve(socket net.Listener) {
+	go func() {
+		_ = rsc.server.Serve(socket)
+	}()
+}
+
+func (rsc *resourcesServerConnector) Stop() {
+	rsc.server.Stop()
+}
+
+func (rsc *resourcesServerConnector) Close(clientConnection *grpc.ClientConn) {
+	_ = clientConnection.Close()
+}
+
+func (rsc *resourcesServerConnector) Register(client pluginapi.RegistrationClient, reqt *pluginapi.RegisterRequest) error {
+	_, err := client.Register(context.Background(), reqt)
+	return err
 }
 
-// NewResourceServer returns an initialized resourceServer
+func (rsc *resourcesServerConnector) Dial(unixSocketPath string, timeout time.Duration) (*grpc.ClientConn, error) {
+	var c *grpc.ClientConn
+	var err error
+	connChannel := make(chan interface{})
+
+	ctx, timeoutCancel := context.WithTimeout(context.TODO(), timeout)
+	defer timeoutCancel()
+	go func() {
+		c, err = grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(),
+			grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+				return net.Dial("unix", addr)
+			}),
+		)
+		connChannel <- "done"
+	}()
+
+	select {
+	case <-ctx.Done():
+		return nil, fmt.Errorf("timout while trying to connect %s", unixSocketPath)
+
+	case <-connChannel:
+		return c, err
+	}
+}
+
+// newResourceServer returns an initialized server
```

Comment:
is this a package local method ? i see the comment with capital 'N' and the method itself with lowercase 'n'

Reply:
fixed

Reply:
fixed

Reply:
fixed